pkgs/applications/virtualization/singularity/packages.nix

```diff
@@ -0,0 +1,54 @@
+{ lib
+, callPackage
+, fetchurl
+}:
+
+{
+  singularity-legacy = (callPackage ./template.nix rec {
+    pname = "singularity-legacy";
+    version = "3.8.6";
+    projectName = "singularity";
+
+    src = fetchurl {
+      url = "https://github.com/hpcng/singularity/releases/download/v${version}/singularity-${version}.tar.gz";
+      sha256 = "sha256-u1o7dnCsnHpLPOWyyfPWtb5g4hsI0zjJ39q7eyqZ9Sg=";
+    };
+  }).overrideAttrs (oldAttrs: {
+    meta = oldAttrs.meta // (with lib; {
+      description = oldAttrs.meta.description + " (leagacy version before renaming)";
```

Comment:
I don't know why  `meta.description` needs to be modified, but if it needs to be: perhaps you could instead use `optionalString (projectName == "singularity")` in `template.nix`. It might turn out cleaner than `overrideAttrs` 🤔

Reply:
Just want to make them distinguishable when performing `nix search`. I can simply remove it if it sounds redundant or over-engineered.

As for the `projectName`, it is used to tell if the singularity package is called `singularity` or `apptainer`. It cannot be used to distinguish between `singularity-legacy` and `singularity-ce`, which is maintained by different organizations.

Reply:
I honestly don't know. Maybe even pass the description as an argument into the template derivation?